src/main/scala/cromwell/Main.scala

```diff
@@ -21,167 +20,184 @@ object Actions extends Enumeration {
+        argOptions(idx) orElse default filterNot (_ == "-") map wdlPath.resolveSibling
+
+
+      // Inputs will be assumed to be at ".json" (unless overridden like all paths with a "-")
+      val inputsPath = argPath(1, Option(swapWdl(".json")))
+      // We'll check if the ".options.json" file exists and use it
+      val workflowOptionsPath = argPath(2, Option(swapWdl(".options.json")).filter(f => Files.exists(Paths.get(f))))
+      // No default for the metadata
+      val workflowMetadataPath = argPath(3, None)
 
-    args foreach { arg =>
-      val path = Paths.get(arg)
-      if (arg != "-") {
+      Log.info(s"Default backend: ${WorkflowManagerActor.BackendType}")
+      Log.info(s"RUN sub-command")
+
+      /* Utilities for the mini-DSL below when testing files during trySources. */
+
+      /** Read the path to a string. Also, as the first time accessing the file, log what we're accessing.  */
+      def readContent(inputDescription: String, path: Path): String = {
+        Log.info(s"  $inputDescription: $path")
+        // Provide slightly better error messages than just going the reading the content.
         if (!Files.exists(path)) {
-          System.err.println(s"ERROR: file does not exist: $arg")
-          System.exit(1)
+          throw new RuntimeException(s"$inputDescription does not exist: $path")
         }
         if (!Files.isReadable(path)) {
-          System.err.println(s"ERROR: file is not readable: $arg")
-          System.exit(1)
+          throw new RuntimeException(s"$inputDescription is not readable: $path")
         }
+        path.contentAsString
       }
-    }
 
-    val wdlFile = args(0)
-    val inputsJsonFile = Try(args(1)).getOrElse(wdlFile.replaceAll("\\.wdl$", ".json"))
-    val workflowOptionsFile = Try(args(2)).getOrElse(wdlFile.replace("\\.wdl$", ".options.json"))
-
-    Log.info(s"Default backend: ${WorkflowManagerActor.BackendType}")
-    Log.info(s"RUN sub-command")
-    Log.info(s"  WDL file: ${args(0)}")
+      /** Read the path to a string, unless the path is None, in which case returns "{}". */
+      def readJson(inputDescription: String, pathOption: Option[Path]): String = {
+        pathOption match {
+          case Some(path) => readContent(inputDescription, path)
+          case None => "{}"
+        }
+      }
 
-    try {
-      val wdlSource = Paths.get(args(0)).slurp
-      val inputsJson = inputsJsonFile match {
-        case "-" => "{}"
-        case path if path != args(0) && Files.exists(Paths.get(path)) =>
-          Log.info(s"  Inputs: $path")
-          Paths.get(path).slurp
-        case _ =>
-          System.err.println(s"ERROR: No workflow inputs specified")
-          System.exit(1)
-          ""
+      /**
+       * Try to write to the path by appending a blank string to file.
+       * Also, as the first time accessing the file, log what we're accessing.
+       */
+      def touchPath(outputDescription: String, pathOption: Option[Path]): Unit = {
```

Comment:
Their `touch` modifies the date on the file, it doesn't test writing. I'll rename mine.